### PR TITLE
update UUV and rover controllers to use trajectory_setpoint and cleanup unused position_setpoint fields

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: src/modules/rover_pos_control/RoverPositionControl.hpp [hunk 1]
```diff
@@ -155,6 +158,11 @@ class RoverPositionControl final : public ModuleBase<RoverPositionControl>, publ
 	/* previous waypoint */
 	matrix::Vector2d _prev_wp{0, 0};
 
+	enum class VelocityFrame {
+		NED,
+		BODY,
+	} _velocity_frame{VelocityFrame::NED};
```
Comment (Jaeyoung-Lim) [inline src/modules/rover_pos_control/RoverPositionControl.hpp:164]: Why are the frame specific for the rover pos control?

**Review by Jaeyoung-Lim — COMMENTED** [review]
Thanks! looks good for me!

Just one minor comment on the velocity frame in rover pos control

File: src/modules/rover_pos_control/RoverPositionControl.hpp (hunk 1 above)
Comment (dagar) [reply src/modules/rover_pos_control/RoverPositionControl.hpp:164]: It's not handled yet, but I'm assuming we're going to add it. Right now ORB_ID(trajectory_setpoint) is always NED, with things like body frame velocity setpoints being converted by the mavlink receiver. I'd like to push all of that into the controllers and keep mavlink as a dumb passthrough.

**Review by MaEtUgR — APPROVED** [review]
Assuming `trajectory_setpoint` is set like the few fields of the triplet before it looks good to me 👍

**Pull request merged**